internal/cli/status.go

```diff
@@ -0,0 +1,845 @@
+
+	if len(cmdArgs) > 1 {
+		c.ui.Output("No more than 1 argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Determine which view to show based on user input
+	var projectTarget, appTarget string
+	if len(cmdArgs) >= 1 {
+		s := cmdArgs[0]
+		target := strings.Split(s, "/") // This breaks if we allow projects with "/" in the name
+
+		projectTarget = target[0]
+		if len(target) == 2 {
+			appTarget = target[1]
+		}
+
+	} else if len(cmdArgs) == 0 {
+		// If we're in a project dir, load the name. Otherwise we'll
+		// show a list of all projects and their status and leave projectTarget
+		// blank
+		if c.project.Ref() != nil {
+			projectTarget = c.project.Ref().Project
+		}
+	}
+
+	if appTarget == "" && c.flagApp != "" {
+		appTarget = c.flagApp
+	} else if appTarget != "" && c.flagApp != "" {
+		// setting app target and passing the flag app is a collision
+		c.ui.Output(wpAppFlagAndTargetIncludedMsg, terminal.WithWarningStyle())
+	}
+
+	// Generate a status view
+	if projectTarget == "" || c.flagAllProjects {
+		// Show high-level status of all projects
+		err = c.FormatProjectStatus()
+		if err != nil {
+			c.ui.Output("CLI failed to build project statuses:", terminal.WithErrorStyle())
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+	} else if projectTarget != "" && appTarget == "" {
+		// Show status of apps inside project
+		err = c.FormatProjectAppStatus(projectTarget)
+		if err != nil {
+			if status.Code(err) == codes.NotFound {
+				c.ui.Output(wpProjectNotFound, projectTarget, c.serverCtx.Server.Address, terminal.WithErrorStyle())
+			} else {
+				c.ui.Output("CLI failed to format project app statuses:", terminal.WithErrorStyle())
+				c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			}
+			return 1
+		}
+	} else if projectTarget != "" && appTarget != "" {
+		// Advanced view of a single app status
+		err = c.FormatAppStatus(projectTarget, appTarget)
+		if err != nil {
+			if status.Code(err) == codes.NotFound {
+				c.ui.Output(wpAppNotFound, appTarget, projectTarget, c.serverCtx.Server.Address, terminal.WithErrorStyle())
+			} else {
+				c.ui.Output("CLI failed to format app status", terminal.WithErrorStyle())
+				c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			}
+			return 1
+		}
+	}
+
+	return 0
+}
+
+// FormatProjectAppStatus formats all applications inside a project
+func (c *StatusCommand) FormatProjectAppStatus(projectTarget string) error {
+	if !c.flagJson {
+		c.ui.Output(wpStatusProjectMsg, projectTarget, c.serverCtx.Server.Address)
+	}
+
+	// Get our API client
+	client := c.project.Client()
+
+	resp, err := client.GetProject(c.Ctx, &pb.GetProjectRequest{
+		Project: &pb.Ref_Project{
+			Project: projectTarget,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	project := resp.Project
+
+	var workspace string
+	if len(resp.Workspaces) == 0 {
+		// this happens if you just wapyoint init
+		workspace = "???"
```

Comment:
I would suggesting using `nil`. Althogh rare, I can see that somebody could eventually use `???` as a workspace and break this part 😅

Reply:
Definitely! Good call.

Reply:
Hm, this seems like odd behavior, in that after an `init` we should know the workspace? And in fact we know it if we specify the app.

```
$ waypoint status
Current status for project "example-go" in server context "localhost:9701".

APP       	WORKSPACE	LATEST STATUS	LAST CHECK 
example-go	default  	N/A          	          	

...
$ waypoint status -app=example-go
Current status for application "example-go" in project "example-go" in server context "localhost:9701".

Application Summary
APP       	WORKSPACE	LATEST STATUS	LAST CHECK 
example-go	???      	N/A          	          	
```

Reply:
Good catch @krantzinator ! That behavior was from me trying to be helpful if worksace was unset and just "try" default. But I think we can rely on the API here to do the right thing if we pass it as unset so I'll remove it.